app/src/main/java/blockeq/com/stellarwallet/activities/InflationActivity.kt

```diff
@@ -23,19 +26,33 @@ class InflationActivity : BaseActivity() {
         addressEditText.setText(Constants.INFLATION_DESTINATION)
 
         saveButton.setOnClickListener {
-            if (NetworkUtils(this).isNetworkAvailable()) {
-                Horizon.Companion.JoinInflationDestination(object : SuccessErrorCallback {
-                    override fun onSuccess() {
-                        Toast.makeText(this@InflationActivity, "Inflation destination set!", Toast.LENGTH_SHORT).show()
-                    }
+            launchPINView(PinType.CHECK, "", "", false)
+        }
+    }
 
-                    override fun onError() {
-                        Toast.makeText(this@InflationActivity, "There was an error setting inflation destination.", Toast.LENGTH_SHORT).show()
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == PinActivity.PIN_REQUEST_CODE) {
+            when (resultCode) {
+                Activity.RESULT_OK -> {
+                    val secretSeed = data!!.getCharArrayExtra(PinActivity.KEY_SECRET_SEED)
+
+                    if (NetworkUtils(this).isNetworkAvailable()) {
+                        Horizon.Companion.JoinInflationDestination(object : SuccessErrorCallback {
+                            override fun onSuccess() {
+                                Toast.makeText(this@InflationActivity, getString(R.string.inflation_set_success), Toast.LENGTH_SHORT).show()
```

Comment:
We can also close/dismiss/pop activity after successful addition of inflation

Reply:
@MeghaBambra added finish() onSuccess!